### ISSUE TITLE
NPE within Transaction

### ISSUE THREAD

**Issue by n3integration** [orientechnologies/orientdb#5210]
I have experienced the following exception a few times while running OrientDB within a transactional context when inserting data.

```java
java.lang.NullPointerException: null
	at com.orientechnologies.orient.core.index.hashindex.local.cache.OWOWCache.fileNameById(OWOWCache.java:865) ~[orientdb-core-2.1-rc5.jar:2.1-rc5]
	at com.orientechnologies.orient.core.index.hashindex.local.cache.OReadWriteDiskCache.fileNameById(OReadWriteDiskCache.java:236) ~[na:na]
	at com.orientechnologies.orient.core.storage.impl.local.paginated.atomicoperations.OAtomicOperation.fileNameById(OAtomicOperation.java:281) ~[orientdb-core-2.1-rc5.jar:2.1-rc5]
	at com.orientechnologies.orient.core.db.record.ridbag.sbtree.OIndexRIDContainerSBTree.<init>(OIndexRIDContainerSBTree.java:60) ~[orientdb-core-2.1-rc5.jar:2.1-rc5]
	at com.orientechnologies.orient.core.db.record.ridbag.sbtree.OIndexRIDContainer.convertToSbTree(OIndexRIDContainer.java:250) ~[orientdb-core-2.1-rc5.jar:2.1-rc5]
	at com.orientechnologies.orient.core.db.record.ridbag.sbtree.OIndexRIDContainer.checkTopThreshold(OIndexRIDContainer.java:221) ~[orientdb-core-2.1-rc5.jar:2.1-rc5]
	at com.orientechnologies.orient.core.db.record.ridbag.sbtree.OIndexRIDContainer.add(OIndexRIDContainer.java:160) ~[orientdb-core-2.1-rc5.jar:2.1-rc5]
	at com.orientechnologies.orient.core.db.record.ridbag.sbtree.OIndexRIDContainer.add(OIndexRIDContainer.java:43) ~[orientdb-core-2.1-rc5.jar:2.1-rc5]
	at com.orientechnologies.orient.core.index.OIndexMultiValues.putInSnapshot(OIndexMultiValues.java:204) ~[orientdb-core-2.1-rc5.jar:2.1-rc5]
	at com.orientechnologies.orient.core.index.OIndexAbstract.applyIndexTxEntry(OIndexAbstract.java:1117) ~[orientdb-core-2.1-rc5.jar:2.1-rc5]
	at com.orientechnologies.orient.core.index.OIndexAbstract.addTxOperation(OIndexAbstract.java:749) ~[orientdb-core-2.1-rc5.jar:2.1-rc5]
	at com.orientechnologies.orient.core.tx.OTransactionOptimistic$CommitIndexesCallback.run(OTransactionOptimistic.java:98) ~[orientdb-core-2.1-rc5.jar:2.1-rc5]
	at com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.commit(OAbstractPaginatedStorage.java:924) ~[orientdb-core-2.1-rc5.jar:2.1-rc5]
	at com.orientechnologies.orient.core.tx.OTransactionOptimistic.doCommit(OTransactionOptimistic.java:488) ~[orientdb-core-2.1-rc5.jar:2.1-rc5]
	at com.orientechnologies.orient.core.tx.OTransactionOptimistic.commit(OTransactionOptimistic.java:147) ~[orientdb-core-2.1-rc5.jar:2.1-rc5]
	at com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.commit(ODatabaseDocumentTx.java:2437) ~[orientdb-core-2.1-rc5.jar:2.1-rc5]
	at com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.commit(ODatabaseDocumentTx.java:2407) ~[orientdb-core-2.1-rc5.jar:2.1-rc5]
	at com.orientechnologies.orient.server.network.protocol.binary.ONetworkProtocolBinary.commit(ONetworkProtocolBinary.java:1092) ~[orientdb-server-2.1-rc5.jar:2.1-rc5]
	at com.orientechnologies.orient.server.network.protocol.binary.ONetworkProtocolBinary.executeRequest(ONetworkProtocolBinary.java:394) ~[orientdb-server-2.1-rc5.jar:2.1-rc5]
	at com.orientechnologies.orient.server.network.protocol.binary.OBinaryNetworkProtocolAbstract.execute(OBinaryNetworkProtocolAbstract.java:217) ~[orientdb-server-2.1-rc5.jar:2.1-rc5]
	at com.orientechnologies.common.thread.OSoftThread.run(OSoftThread.java:69) ~[orientdb-core-2.1-rc5.jar:2.1-rc5]
```

**Comment by lvca** [comment]
Could you please don't use RC but stable releases? Try 2.1.4.

**Comment by n3integration** [comment]
@lvca - good point. I've upgraded my environment to use the 2.1.4 release of orientdb. I am getting the same stack trace when attempting to commit a transaction.

**Comment by lvca** [comment]
@laa could you take this please?

**Comment by laa** [comment]
It can not be the same exception because code is changed radically you should provide new one. 
Did you do export/import from to JSON or merely upgraded jar ? Problems with OIndexContainer is already fixed and tested by one of our clients but you have to do export/import on your side. If issue still persists please provide up to date version of exception.

**Comment by n3integration** [comment]
@laa - I only upgraded the client jar and left the server intact. With that said, the server was running a more recent version than the client. I did receive a warning in the log file, but it appeared harmless. I received a stack trace that at least appeared to be very similar to the above after the upgrade, but after a week, we have yet to experience the issue again. I apologize for the late reply. I wanted to make certain that we were no longer experiencing the issue before responding. I think that it is safe to close out the issue as the upgrade of the client seems to have corrected the behavior.
